ambassador/ambassador/ir/irbasemappinggroup.py

```diff
@@ -9,4 +9,54 @@
 
 
 class IRBaseMappingGroup (IRResource):
-    pass
+    def normalize_weights_in_mappings(self):
```

Comment:
How much of this is moving code and how much is new?

Reply:
This is about 95% new - and is the meat of the PR.

The major difference is -
- till now we've wanted the weights to total to 100, e.g. 10 + 20 + 30 + 40.
- but now we want weights to increment to 100. So if we want assigned weights to be 10, 20, 30, 40 - we want feed the following to runtime fraction - 10, 30, 60, 100, i.e. 10, 10+20, 10+20+30, 10+20+30+40.

Like you had explained to me earlier, runtime fraction is going to calculate a number between 0 and 100, and whichever bracket it falls into, it's going to route to that.

So, if this normalizing algorithm goes wrong, weights are gonna break.